### PR TITLE
Reading history file fails with non-latin chars on Windows

### PR DESCRIPTION
**Pull request by ankostis**
A band-aid PR that fixes pyreadline/pyreadline#49 and pyreadline/pyreadline/#55 on non-latin commands in history-file.

### CODE REVIEW

**Comment by eabase** [comment]
oh, this one look tricky! I know @brgirgis already did some fixes in just about these same places. 

@ankostis  
Can you explain what the `encoding=pyreadline_codepage, errors='replace'` parts does? (And why they are needed?)
I've been preaching to just consistently use `utf-8`, so what is `pyreadline_codepage` doing? I think we need to be careful with trying to outsmart what our Python, OS and terminals use for code pages.

**Comment by ankostis** [comment]
The `_pyreadline_codepage` is the fallback encoding "constant" set to `utf-8` as you said - it's just there not to duplicate references to the string `"utf-8" in the code, and provide an opportunity for discrepancies, 
to signify the developer that this should be the same encoding used throughout all these places.

The "replace" error handling is a safety measure not to crash in case of invalid utf-8 sequences.

**Comment by ankostis** [comment]
I've also had written the above this in the docstring of the "private" constant.

File: readline.py [hunk 1]
```diff
@@ -24,6 +24,7 @@
     'get_completer',
     'get_begidx',
     'get_endidx',
+    'redisplay',
```
Comment (brgirgis) [inline readline.py:27]: This one already exists at the end of the list.

File: pyreadline3/unicode_helper.py [hunk 2]
```diff
@@ -8,15 +8,17 @@
 # *****************************************************************************
 import sys
 
+#: Also support non-latin chars.
```
Comment (brgirgis) [inline pyreadline3/unicode_helper.py:11]: nice!

**Comment by eabase** [comment]
@ankostis  
Ok, I see what you're doing. Generally a good idea. However, here I am a bit ambivalent as it make it less clear when looking at the code, *what* the character-set actually is. At this point I am not sure what is better. I'm usually for code clarity/transparency, but if someone (god forbid) need to change the default character-set to something different than UTF-8 then it would be good. However, i think that is a fringe case at this point.

**Comment by ankostis** [comment]
I didn't chang the fall-back logic - simply changed it from ASCI -> UTF-8.  I believe you agree ion that.

Additionally replaced read/write history-files, from binary --> text when opening them, in a context-manager.
It would be nice to review the codebase and apply the same fix & enslavement to all cases where files are read.

Finally, i introduced an intermediate fall-back variable, used when opening textual files. 
I agree with you, that was a mistake.
Better leave default encodins when reading writing textual files.

I will push a fix commit able the last point.

**Comment by ankostis** [comment]
Done.
And blindly applied the same (default encoding) for al str.conversion utility functions.

You still have to aprove me, for the CI workflow to kick in.

**Comment by eabase** [comment]
@ankostis  You still didn't fix the typo I first reported above.

**Comment by ankostis** [comment]
I'm cluless to where your report is?  Did i react to it?  ... have no recollection of it.

**Comment by eabase** [comment]
@ankostis 
At the top of this page in my review of line 12.  😄

**Comment by ankostis** [comment]
I can't see anything at line 12.
Could it be at a rewritten commit, and hence lost?

**Comment by eabase** [comment]
@ankostis 
Ok, I see the problem. It's wicked that my reviews are only seen by myself!! 
Check your typo here:
* https://github.com/pyreadline3/pyreadline3/blob/8a0ed94b977a4a2dd8c6ea05d42a19ed30c87c31/pyreadline3/unicode_helper.py#L12

**Comment by ankostis** [comment]
Thank you, fixed it.
Bad that no tests captured this (but it's hard to enable the fallback encoding).

Regarding your review-comment, the comments accumulate until you submit the review, you're sure you had clicked this?

**Comment by ankostis** [comment]
Again its needs your workflow-approval;  could it be that i'm rewriting commits, instead of appending?

**Comment by eabase** [comment]
@ankostis 
Yeah, I never used the review before and there is nothing for me to click here. 
I believe the "maintainer" of this repo  @brgirgis  could involve himself a little bit more... or at least consider giving us some more rights do do what's necessary to help maintain it.

I see you force pushed, which is normally not a good idea as it messes up the git for everyone else. 

In addition Bassem should probably create a dev branch for these things, so that things can get merged and tested more easily without breaking stuff. Otherwise I have to clone also your fork... 

I've had quite enough of this pyreadline stuff, and I just wanted to get some small things working...not ending up maintaining and riposting issues, lists and reporting back/forth with almost zero feedback. (*Should I assume nobody uses pyreadline anymore, or just don't GF about contributing?*)

**Comment by eabase** [comment]
@brgirgis 
Please accept my review, so they can be visible to everyone. 
Can you also give us some feedback on the PR's?

**Comment by brgirgis** [comment]
Please add a couple of test cases to cover this change.

**Pull request merged**